### PR TITLE
Fix issues in makefile dependencies and replace Travis CI with GitHub Actions

### LINKED ISSUE #101: Proposal: replace TravisCI with Github Actions
Since TravisCI changed ownership some months ago, the support for open source projects has dropped significantly. Therefore I suggest to replace TravisCI with Github Actions.

### PR DESCRIPTION
**Pull request by magnusbaeck**
The makefile problems were discovered while trying out the new GitHub Actions workflow. Fixes #101.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: test
+
+on:
+  - push
+  - pull_request
```
Comment (breml) [inline .github/workflows/test.yml:5]: With this configuration, the pipelines are executed twice for branches in this repo that are used to create a PR (as it is the case for this particular PR / branch `actions`).

Based on https://docs.github.com/en/actions/reference/events-that-trigger-workflows, I suggest:

```suggestion
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```

**Review by breml — APPROVED** [review]
LGTM, I made a suggestion to reduce duplicate executions of pipelines.

I look forward to run the integration tests in Github actions as well.

File: .github/workflows/test.yml (hunk 1 above)
Comment (magnusbaeck) [reply .github/workflows/test.yml:5]: Okay, that's fair. Is my understanding correct that workflows are branch-specific, i.e. we'd have to cherry-pick this commit to the maint-1.x branch and replace the master branch reference with maint-1.x?

Comment (breml) [reply .github/workflows/test.yml:5]: Yes, this is correct. We need to apply a similar commit to the `maint-1.x` branch where we replace `master` with `maint-1.x`.

Comment (magnusbaeck) [reply .github/workflows/test.yml]: Thanks for the confirmation. Using your suggestion and pushing an additional commit to the maint-1.x branch.

**Pull request merged**